autoendpoint/src/headers/vapid.rs

```diff
@@ -1,9 +1,67 @@
-use crate::headers::util::split_key_value;
 use std::collections::HashMap;
+use std::fmt;
+
+use base64::Engine;
+use serde::{Deserialize, Serialize};
 use thiserror::Error;
 
+use crate::headers::util::split_key_value;
+use autopush_common::util::{sec_since_epoch, ONE_DAY_IN_SECONDS};
+
 pub const ALLOWED_SCHEMES: [&str; 3] = ["bearer", "webpush", "vapid"];
 
+#[derive(Clone, Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct VapidClaims {
+    pub exp: u64,
+    pub aud: String,
+    pub sub: String,
+}
+
+impl Default for VapidClaims {
+    fn default() -> Self {
+        Self {
+            exp: VapidClaims::default_exp(),
+            aud: "No audience".to_owned(),
+            sub: "No sub".to_owned(),
+        }
+    }
+}
+
+impl VapidClaims {
+    pub fn default_exp() -> u64 {
```

Comment:
Why is this a method vs a const?

Reply:
I feel it's clearer what the value is in the code, since we use the pattern frequently.
I can add a `#[inline(always)]` tag if you like, but I think the compiler will do that for us.